assertk/src/commonTest/kotlin/test/assertk/assertions/CharSequenceTest.kt

```diff
@@ -270,4 +270,18 @@ class CharSequenceTest {
         assertEquals("expected to match:${show(regex)} but was:<\"12345\">", error.message)
     }
     //endregion
+
+    //region contains match
+    @Test fun contains_match_matching_value_passes() {
+        assertThat("abc1234xyz").containsMatch(Regex("\\d\\d\\d\\d"))
+    }
+
+    @Test fun contains_match_not_matching_value_fails() {
+        val regex = Regex("\\d\\d\\d\\d")
+        val error = assertFailsWith<AssertionError> {
+            assertThat("abc123xyz").containsMatch(regex)
+        }
```

Comment:
It's a fun data point for #453 that the tests within the library use `assertFailsWith`!

Reply:
heh, that's because I didn't want to write the assertions with itself as that felt a little circular...